### PR TITLE
fixed error using NNC with thermal

### PR DESCRIPTION
**Pull request by hnil**
* fixed error using NNZ with thermal

also:
* fixed but not checked for diffusion
* bug for zero thermal diffusion (nan in derivatives)
* added checking of consistency between input and taged phases

### CODE REVIEW

**Comment by GitPaean** [comment]
jenkins build this please

File: ebos/ecltransmissibility_impl.hh [hunk 1]
```diff
@@ -866,13 +874,40 @@ applyNncToGridTrans_(const std::unordered_map<std::size_t,int>& cartesianToCompr
...
+        // if (enableDiffusivity_) {
+        //     auto candidate = diffusivity_.find(isId(low, high));
+        //     if (candidate != trans_.end()) {
+        //         // NNC is represented by the grid and might be a neighboring connection
+        //         // In this case the transmissibilty is added to the value already
+        //         // set or computed.
+        //         candidate->second += nncEntry.transDiffusion;
+        //     }
+        // }
     }
```
Comment (blattms) [inline ebos/ecltransmissibility_impl.hh:911]: maybe remove this commented code to clean up?

**Comment by hnil** [comment]
Should we add the test case?

**Review by blattms — APPROVED** [review]
Looks good to me. but I'll leave the floor to the requested reviewers.

**Comment by GitPaean** [comment]
> Should we add the test case?

Probably we can. But we should ask for permission.

File: ebos/ecltransmissibility_impl.hh (hunk 1 above)
Comment (hnil) [reply ebos/ecltransmissibility_impl.hh:911]: I just added it to show where to ad properties if one need it for diffusion and energy and it is added in input + nncEntry.

**Comment by hnil** [comment]
To make a test one only need to add NNC between first and last cell in the CO2STORE test.

File: ebos/ecltransmissibility_impl.hh (hunk 1 above)
Comment (blattms) [reply ebos/ecltransmissibility_impl.hh:911]: I see.
Indeed, NNC's are treated differently in OPM depending on their direction. In Z-direction (connections over pinched out cells) NNCs are repesented as normal faces. In this case Energy and Diffusion will be computed for them. For all explicitly specified NNCs this is not the case and we would need an additional custom keyword.

**Comment by blattms** [comment]
I am merging this. The fix is good and adding a test can be done in a separate PR.

I am 100% sure that we have permission to derive extra tests from existing ones in opm-tests. That is covered by the license.

**Pull request merged**